test/e2e/lifecycle/kubelet_security.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// Try all the proxy tests this many times (to catch even rare flakes).
+	proxyAttempts = 10
+	// Only print this many characters of the response (to keep the logs
+	// legible).
+	maxDisplayBodyLen = 100
+
+	// We have seen one of these calls take just over 15 seconds, so putting this at 30.
+	proxyHTTPCallTimeout = 30 * time.Second
+)
+
+var _ = SIGDescribe("Ports Security Check [Feature:KubeletSecurity]", func() {
+	options := framework.FrameworkOptions{
+		ClientQPS: -1.0,
+	}
+	f := framework.NewFramework("kubelet-security", options, nil)
+
+	prefix := "/api/v1"
+	// make sure kubelet readonly (10255) and cadvisor (4194) ports are disabled via API server proxy
+	It("should not be able to contact the readonly port 10255", func() { portProxyDisabledTest(f, prefix+"/nodes/", ":10255/pods/") })
```

Comment:
suggest: `should not be able to proxy to the readonly kubelet port 10255 using proxy subresource` to be consistent with what you have below